File: src-tauri/src/commands/logs.rs
use crate::models::result::CommandResult;
use crate::services::log_service::{self, ExportDiagnosticsData, LogSource, ReadLogsData};

#[tauri::command]
pub async fn read_logs(source: LogSource, lines: Option<usize>) -> CommandResult<ReadLogsData> {
    match log_service::read_logs(source, lines.unwrap_or(1200)).await {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

#[tauri::command]
pub async fn export_diagnostics(
    source: LogSource,
    keyword: Option<String>,
    content: String,
    archive: Option<bool>,
) -> CommandResult<ExportDiagnosticsData> {
    match log_service::export_diagnostics(source, keyword, content, archive.unwrap_or(false)).await
    {
        Ok(data) => CommandResult::ok(data),
        Err(error) => CommandResult::err(error),
    }
}

File: src/main/ipc/logs.ts
import { ipcMain } from "electron";
import { IPC } from "../../shared/ipc-channels.js";
import { ok, err } from "../models/result.js";
import { toAppError } from "./utils.js";
import { readLogs, exportDiagnostics, LogSource } from "../services/log-service.js";

export function registerLogsHandlers(): void {
  ipcMain.handle(IPC.READ_LOGS, async (_event, payload: { source: string; lines?: number }) => {
    try {
      const source = payload.source as LogSource;
      return ok(await readLogs(source, payload.lines ?? 1200));
    }
    catch (error) { return err(toAppError(error)); }
  });

  ipcMain.handle(IPC.EXPORT_DIAGNOSTICS, async (_event, payload: {
    source: string;
    keyword?: string;
    content: string;
    archive?: boolean;
  }) => {
    try {
      const source = payload.source as LogSource;
      return ok(await exportDiagnostics(source, payload.keyword, payload.content, payload.archive ?? false));
    }
    catch (error) { return err(toAppError(error)); }
  });
}
